modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -423,18 +449,263 @@ public Response blacklistApplications(@HeaderParam("Accept") String acceptHeader
 		Message message = new Message();
 
 		try {
-			ConfigOperation configOperation = new ConfigOperation();
-			configOperation.setCode(AndroidConstants.OperationCodes.BLACKLIST_APPLICATIONS);
-			configOperation.setType(Operation.Type.CONFIG);
+			BlacklistApplications blacklistApplications = blacklistApplicationsBeanWrapper.getOperation();
+
+			if (blacklistApplications == null) {
+				throw new OperationManagementException("Blacklist applications bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.BLACKLIST_APPLICATIONS);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(blacklistApplications.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(blacklistApplicationsBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
+
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (DeviceManagementException e) {
+			message.setResponseMessage("Issue in retrieving device management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
```

Comment:
while the response is built according to the content-type specified by the accept-header, can we just limit the response to JSON by specifying "@Produces(MediaType.APPLICATION_JSON)"?